### PR TITLE
chore: drop python 3.7 for cibuildwheel driver

### PR DESCRIPTION
**Pull request by mayeut**
This drops python 3.7 for cibuildwheel.
Building python 3.6 / 3.7 packages is still supported.

### CODE REVIEW

**Review by henryiii — APPROVED** [review]
This is a week early, but it makes sense to do with a minor point bump, which #1525 requires too, so doing them together makes sense.

**Comment by henryiii** [comment]
Could you bump ruff too? Maybe I can from my phone, actually…

**Review by joerick — APPROVED** [review]
Lovely stuff, thank you @mayeut. Let's just hold on merging this for a few days in case we need to get a patch release out in the meantime.

**Comment by henryiii** [comment]
3.7's EoL is today.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -34,9 +34,9 @@ repos:
   rev: v1.3.0
   hooks:
   - id: mypy
-    name: mypy 3.7 on cibuildwheel/
+    name: mypy 3.8 on cibuildwheel/
     exclude: ^cibuildwheel/resources/.*py$
-    args: ["--python-version=3.7"]
+    args: ["--python-version=3.8"]
     additional_dependencies: &mypy-dependencies
```
Comment (henryiii) [inline .pre-commit-config.yaml:40]: FYI, the markdown-it-py cap below can be removed, it was there to support Python 3.7.

Comment (mayeut) [reply .pre-commit-config.yaml:40]: I'd rather be explicit here if it's OK.

Comment (henryiii) [reply .pre-commit-config.yaml:40]: I mean “markdown-it-py<3” can be removed. I can’t comment on the correct line.

Comment (mayeut) [reply .pre-commit-config.yaml:40]: ok, I misunderstood completely.
will do

**Comment by henryiii** [comment]
I'll probably retry #1297 after this goes in.

**Comment by henryiii** [comment]
> for a few days

It's been three weeks. I'd like to get something out before SciPy next week.

**Comment by joerick** [comment]
My apologies, I was holding out for a minor release but had forgotten that #1525 went in a couple of weeks back.

**Pull request merged**